dbms/src/Flash/Mpp/FineGrainedShuffleWriter.cpp

```diff
@@ -57,81 +59,111 @@ void FineGrainedShuffleWriter<StreamWriterPtr>::finishWrite()
     }
 }
 
+template <class StreamWriterPtr>
+void FineGrainedShuffleWriter<StreamWriterPtr>::prepare(const Block & sample_block)
+{
+    /// Materialize sample_block so that header and reserved scatterColumns are full columns
+    /// Because ser/der don't support constant columns now
+    header = sample_block.cloneEmpty();
+    HashBaseWriterHelper::materializeBlock(header);
+    num_columns = header.columns();
+    // fine_grained_shuffle_stream_count is in (0, 1024], and partition_num is uint16_t, so will not overflow.
+    num_bucket = partition_num * fine_grained_shuffle_stream_count;
+    partition_key_containers_for_reuse.resize(collators.size());
+    resetScatterColumns();
```

Comment:
Looks like `resetScatterColumns` is only used in `prepare`, it's more like `initScatterColumns`?

Reply:
Renamed to initScatterColumns